include/ensmallen_bits/problems/logistic_regression_function_impl.hpp

```diff
@@ -98,28 +103,29 @@ typename MatType::elem_type LogisticRegressionFunction<MatType>::Evaluate(
   // We want to minimize this function.  L2-regularization is just lambda
   // multiplied by the squared l2-norm of the parameters then divided by two.
   typedef typename MatType::elem_type ElemType;
+  typedef typename ForwardType<MatType>::brow BaseRowType;
 
   // For the regularization, we ignore the first term, which is the intercept
   // term and take every term except the last one in the decision variable.
   const ElemType regularization = 0.5 * lambda *
-      arma::dot(parameters.tail_cols(parameters.n_elem - 1),
+      dot(parameters.tail_cols(parameters.n_elem - 1),
       parameters.tail_cols(parameters.n_elem - 1));
 
   // Calculate vectors of sigmoids.  The intercept term is parameters(0, 0) and
   // does not need to be multiplied by any of the predictors.
-  const arma::Row<ElemType> sigmoid = 1.0 / (1.0 +
-      arma::exp(-(parameters(0, 0) +
-                parameters.tail_cols(parameters.n_elem - 1) * predictors)));
+  const BaseRowType sigmoid = 1.0 / (1.0 +
+      exp(-(parameters(0, 0) +
+      parameters.tail_cols(parameters.n_elem - 1) * predictors)));
 
   // Assemble full objective function.  Often the objective function and the
   // regularization as given are divided by the number of features, but this
   // doesn't actually affect the optimization result, so we'll just ignore those
   // terms for computational efficiency.  Note that the conversion causes some
   // copy and slowdown, but this is so negligible compared to the rest of the
   // calculation it is not worth optimizing for.
-  const ElemType result = arma::accu(arma::log(1.0 -
-      arma::conv_to<arma::Row<ElemType>>::from(responses) + sigmoid %
-      (2 * arma::conv_to<arma::Row<ElemType>>::from(responses) - 1.0)));
+  const ElemType result = accu(log(1.0 -
+      conv_to<BaseRowType>::from(responses) + sigmoid %
+      (2 * conv_to<BaseRowType>::from(responses) - 1.0)));
```

Comment:
Just as a side note, unrelated to this PR, this function is way faster if we convert to `BaseRowType` in the constructor, just once, and then use the converted version in the `Evaluate()`/`Gradient()` implementations.